### PR TITLE
(batch)point get use individual gRPC connections and different batch policy

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
### What problem does this PR solve? 

Make super batch works better in `rpcClient`.

### What is changed and how it works?

- Point get use individual gRPC connections, and set the default value to 2, benchmark shows the `sysbench` point select has 8% improvement.
- Point get use different batch threshold
- Set gRPC connections default value to 4

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository

Release note

 - Write release note for bug-fix or new feature.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
/bench
```
tidb.yml:
  tikv_client:
    # Max gRPC connections that will be established with each tikv-server.
    grpc-connection-count: 4
```

**Comment by mahjonp** [comment]
/bench
```
tidb.yml:
  tikv_client:
    # Max gRPC connections that will be established with each tikv-server.
    grpc-connection-count: 4
```

**Comment by mahjonp** [comment]
/bench
```
tidb.yml:
  tikv_client:
    # Max gRPC connections that will be established with each tikv-server.
    grpc-connection-count: 4
```

File: store/tikv/client.go [hunk 1]
```diff
@@ -224,36 +234,58 @@ func NewTestRPCClient() Client {
...
+	case Read:
+		array, ok = c.readConns[addr]
+	case Write:
+		array, ok = c.writeConns[addr]
+	}
 	if !ok {
 		var err error
 		connCount := config.GetGlobalConfig().TiKVClient.GrpcConnectionCount
+		if ctype == Read && connCount > 2 {
+			connCount /= 2
```
Comment (tabokie) [inline store/tikv/client.go]: Maybe assign less conns to read? As the results in #12569 shows 3 conn is best for write, 2 conn is best for read.

Comment (zhangjinpeng1987) [reply store/tikv/client.go]: I will make read and write connection count configurable after we have got a proper default value.

**Comment by zhangjinpeng1987** [comment]
/bench
```
tidb.yml:
  tikv_client:
    # Max gRPC connections that will be established with each tikv-server.
    grpc-connection-count: 4
```

File: store/tikv/client.go (hunk 1 above)
Comment (coocood) [reply store/tikv/client.go]: If `connCount == 3` , `/=2` gets `1`, which is not optimal.

Comment (zhangjinpeng1987) [reply store/tikv/client.go]: I have make point get connection count configurable, and will give a default value after i have finished performance test.

**Comment by zhangjinpeng1987** [comment]
/bench
```
tidb.yml:
  tikv_client:
    # Max gRPC connections that will be established with each tikv-server.
    grpc-connection-count: 4
    point-get-grpc-connection-count: 2
```

**Comment by tiancaiamao** [comment]
You may need `go fmt ./...` to make CI pass @zhangjinpeng1987

**Comment by tiancaiamao** [comment]
The performance benchmark result is ... decreased? @zhangjinpeng1987

**Pull request closed without merging**